src/highlevel/finite_diff.jl

```diff
@@ -10,6 +10,7 @@ __getfield(c::FiniteDiffJacobianCache, ::Val{:jac_prototype}) = c.jac_prototype
 
 function sparse_jacobian_cache(fd::Union{AutoSparseFiniteDiff, AutoFiniteDiff},
         sd::AbstractMaybeSparsityDetection, f::F, x; fx = nothing) where {F}
+    x = __make_mutable(x) # FiniteDiff is bad at handling immutables
```

Comment:
how so? What is wrong with those dispatches?

Reply:
I fixed it without this. Ig the assumption there is if `J` is mutable, `x` is implicitly assumed to be mutable in FiniteDiff, which is a reasonable assumption.